hydradx/tests/test_omnipool_amm.py

```diff
@@ -1020,57 +1001,58 @@ def test_sell_stableswap_for_stableswap(initial_state: oamm.OmnipoolState):
+# def test_slip_fees(initial_state: oamm.OmnipoolState, lrna_slip_rate: float, asset_slip_rate: float):
+#     initial_state.lrna_fee = oamm.slip_fee(lrna_slip_rate, minimum_fee=0.0001)
+#     initial_state.asset_fee = oamm.slip_fee(asset_slip_rate, minimum_fee=0.0001)
+#     initial_agent = Agent(holdings={tkn: 1000000 for tkn in initial_state.asset_list})
+#     tkn_buy = initial_state.asset_list[2]
+#     tkn_sell = initial_state.asset_list[3]
+#     sell_quantity = 1
+#     sell_state, sell_agent = oamm.swap(initial_state, initial_agent, tkn_buy, tkn_sell, sell_quantity=sell_quantity)
+#     split_sell_state, split_sell_agent = initial_state.copy(), initial_agent.copy()
+#     next_state, next_agent = {}, {}
+#     for i in range(2):
+#         next_state[i], next_agent[i] = oamm.swap(
+#             old_state=split_sell_state,
+#             old_agent=split_sell_agent,
+#             tkn_sell=tkn_sell,
+#             tkn_buy=tkn_buy,
+#             sell_quantity=sell_quantity / 2
+#         )
+#         split_sell_state, split_sell_agent = next_state[i], next_agent[i]
+#     if split_sell_agent.holdings[tkn_buy] < sell_agent.holdings[tkn_buy]:
+#         raise AssertionError('Agent failed to save money by splitting the sell order.')
+#
+#     buy_quantity = 1
+#     buy_state, buy_agent = oamm.swap(initial_state, initial_agent, tkn_buy, tkn_sell, buy_quantity=buy_quantity)
+#     split_buy_state, split_buy_agent = initial_state.copy(), initial_agent.copy()
+#     next_state, next_agent = {}, {}
+#     for i in range(2):
+#         next_state[i], next_agent[i] = oamm.swap(
+#             old_state=split_buy_state,
+#             old_agent=split_buy_agent,
+#             tkn_sell=tkn_sell,
+#             tkn_buy=tkn_buy,
+#             buy_quantity=buy_quantity / 2
+#         )
+#         split_buy_state, split_buy_agent = next_state[i], next_agent[i]
+#     if split_buy_agent.holdings[tkn_sell] < buy_agent.holdings[tkn_sell]:
+#         raise AssertionError('Agent failed to save money by splitting the buy order.')
+#
+#     if ((initial_agent.holdings[tkn_sell] + initial_agent.holdings[tkn_buy]
+#          + initial_state.liquidity[tkn_sell] + initial_state.liquidity[tkn_buy])
+#             != pytest.approx(buy_agent.holdings[tkn_sell] + buy_agent.holdings[tkn_buy]
+#                              + buy_state.liquidity[tkn_sell] + buy_state.liquidity[tkn_buy])):
+#         raise AssertionError('Asset quantity is not constant after trade (one-part)')
+#
+#     if ((initial_agent.holdings[tkn_sell] + initial_agent.holdings[tkn_buy]
+#          + initial_state.liquidity[tkn_sell] + initial_state.liquidity[tkn_buy])
+#             != pytest.approx(split_buy_agent.holdings[tkn_sell] + split_buy_agent.holdings[tkn_buy]
+#                              + split_buy_state.liquidity[tkn_sell] + split_buy_state.liquidity[tkn_buy])):
+#         raise AssertionError('Asset quantity is not constant after trade (two-part)')
 
 
 # TODO re-enable this test
```

Comment:
Can we re-enable this test?

Reply:
hmm, omnipool_reasonable_config was there in the original rebase but then it made me rebase again (no idea why) and it must have gone missing

Reply:
Fixed this and also commented out an old dynamic fees test that was using the old version of dynamic fees.

We should consider writing a test for the new version, and also perhaps renaming.

Reply:
Re-enabled the 2 tests I had disabled. One worked, one I introduced a pytest.approx() into the assert because the test was being done with mpf variables, so the results were not *exactly* identical anymore.